File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/DomainDao.java
package com.platdmit.simplecloudmanager.data.database.dao;

import com.platdmit.simplecloudmanager.data.database.entity.DbDomain;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Query;

@Dao
public interface DomainDao extends BaseDao<DbDomain>{
    @Query("SELECT * FROM dbdomain")
    List<DbDomain> getAll();

    @Query("SELECT * FROM dbdomain WHERE mId = :id")
    DbDomain getElement(long id);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/DomainDao.kt
package com.platdmit.simplecloudmanager.data.database.dao

import androidx.room.Dao
import androidx.room.Query
import com.platdmit.simplecloudmanager.data.database.entity.DbDomain

@Dao
interface DomainDao : BaseDao<DbDomain> {
    @Query("SELECT * FROM dbdomain")
    fun getAllElement(): List<DbDomain>?

    @Query("SELECT * FROM dbdomain WHERE id = :id")
    fun getElement(id: Long): DbDomain?
}